src/DynamoUtilities/CertificateVerification.cs

```diff
@@ -3,12 +3,194 @@
 using System.IO;
 using System.Linq;
 using System.Reflection;
+using System.Runtime.InteropServices;
 using System.Security.Cryptography.X509Certificates;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace DynamoUtilities
 {
+    internal sealed class WinTrustInterop
+    {
+        #region WinTrustData struct field enums
+        enum WinTrustDataUIChoice : uint
+        {
+            All = 1,
+            None = 2,
+            NoBad = 3,
+            NoGood = 4
+        }
+
+        enum WinTrustDataRevocationChecks : uint
+        {
+            None = 0x00000000,
+            WholeChain = 0x00000001
+        }
+
+        enum WinTrustDataChoice : uint
+        {
+            File = 1,
+            Catalog = 2,
+            Blob = 3,
+            Signer = 4,
+            Certificate = 5
+        }
+
+        enum WinTrustDataStateAction : uint
+        {
+            Ignore = 0x00000000,
+            Verify = 0x00000001,
+            Close = 0x00000002,
+            AutoCache = 0x00000003,
+            AutoCacheFlush = 0x00000004
+        }
+
+        [FlagsAttribute]
+        enum WinTrustDataProvFlags : uint
+        {
+            UseIe4TrustFlag = 0x00000001,
+            NoIe4ChainFlag = 0x00000002,
+            NoPolicyUsageFlag = 0x00000004,
+            RevocationCheckNone = 0x00000010,
+            RevocationCheckEndCert = 0x00000020,
+            RevocationCheckChain = 0x00000040,
+            RevocationCheckChainExcludeRoot = 0x00000080,
+            SaferFlag = 0x00000100,        // Used by software restriction policies. Should not be used.
+            HashOnlyFlag = 0x00000200,
+            UseDefaultOsverCheck = 0x00000400,
+            LifetimeSigningFlag = 0x00000800,
+            CacheOnlyUrlRetrieval = 0x00001000,      // affects CRL retrieval and AIA retrieval
+            DisableMD2andMD4 = 0x00002000      // Win7 SP1+: Disallows use of MD2 or MD4 in the chain except for the root 
+        }
+
+        enum WinTrustDataUIContext : uint
+        {
+            Execute = 0,
+            Install = 1
+        }
+        #endregion
+
+        #region WinTrust structures
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        class WinTrustFileInfo
+        {
+            UInt32 StructSize = (UInt32)Marshal.SizeOf(typeof(WinTrustFileInfo));
```

Comment:
is this used anywhere?

Reply:
It is not

Reply:
But it fails if it is not set

Reply:
okay, I assume this is because the the class needs this field to exist when passed into native code and the layout is sequential so it needs to be first.

Reply:
yes, I guess these map to `cbStruct`